cmd/nerdctl/compose_exec_linux_test.go

```diff
@@ -207,3 +211,83 @@ services:
 	base.ComposeCmdWithHelper(unbuffer, "-f", comp.YAMLFullPath(), "exec", "-t=false", "svc0", "stty").AssertFail()                         // `-i`
 	base.ComposeCmdWithHelper(unbuffer, "-f", comp.YAMLFullPath(), "exec", "-i=false", "-t=false", "svc0", "stty").AssertFail()
 }
+
+func TestComposeExecWithIndex(t *testing.T) {
+	testutil.DockerIncompatible(t)
+	base := testutil.NewBase(t)
+	var dockerComposeYAML = fmt.Sprintf(`
+version: '3.1'
+
+services:
+  svc0:
+    image: %s
+    command: "sleep infinity"
+    deploy:
+      replicas: 3
+`, testutil.CommonImage)
+
+	comp := testutil.NewComposeDir(t, dockerComposeYAML)
+	t.Cleanup(func() {
+		comp.CleanUp()
+	})
```

Comment:
what's the difference compared to call `defer comp.CleanUp()` directly?

Reply:
`t.CleanUp` ensures resources clean up after all subtests are done, which is useful when subtests enable `t.Parallel`
FYI: https://github.com/golang/go/issues/37333